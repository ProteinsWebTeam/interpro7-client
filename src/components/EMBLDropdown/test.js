// @flow
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { noop } from 'lodash-es';

import { EMBLDropdown } from '.';

const renderer = new ShallowRenderer();

describe('<EMBLDropdown />', () => {
  test('should render as visible', () => {
    renderer.render(<EMBLDropdown closeEMBLMapNav={noop} visible />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  test('should render as invisible', () => {
    renderer.render(<EMBLDropdown closeEMBLMapNav={noop} />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
