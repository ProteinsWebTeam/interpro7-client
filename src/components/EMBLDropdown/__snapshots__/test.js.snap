// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EMBLDropdown /> should render as invisible 1`] = `
<div
  aria-hidden={true}
  className="masthead-black-bar"
  inert=""
>
  <nav
    className="dropdown-pane bottom embl-dropdown"
    id="embl-dropdown"
  >
    <div
      className="row padding-bottom-medium"
    >
      <div
        className="columns padding-top-medium"
      >
        <button
          aria-label="Close alert"
          className="close-button"
          onClick={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
          >
            ×
          </span>
        </button>
      </div>
      <div
        className="columns medium-7"
      >
        <div
          className="large-10 medium-12"
        >
          <div
            className="margin-bottom-large padding-top-xsmall margin-top-large"
          >
            <h3
              className="no-underline inline"
              style={
                Object {
                  "lineHeight": "1rem",
                }
              }
            >
              <Connect(Link)
                href="https://embl.org"
              >
                EMBL
              </Connect(Link)>
            </h3>
             
            was set up in 1974 as Europe’s flagship laboratory for the life sciences – an intergovernmental organisation with more than 80 independent research groups covering the spectrum of molecular biology:
          </div>
        </div>
        <div
          className="row large-up-3 medium-up-3 small-up-2 no-underline medium-11"
        >
          <div
            className="column padding-bottom-medium"
          >
            <Connect(Link)
              href="https://www.embl.de/research/index.php"
            >
              <h5
                className="inline underline"
              >
                Research:
              </h5>
               
              perform basic research in molecular biology
            </Connect(Link)>
          </div>
          <div
            className="column padding-bottom-medium"
          >
            <Connect(Link)
              href="https://www.embl.de/services/index.html"
            >
              <h5
                className="inline underline"
              >
                Services:
              </h5>
               
              offer vital services to scientists in the member states
            </Connect(Link)>
          </div>
          <div
            className="column padding-bottom-medium"
          >
            <Connect(Link)
              href="https://www.embl.de/training/index.php"
            >
              <h5
                className="inline underline"
              >
                Training
              </h5>
               
              scientists, students and visitors at all levels
            </Connect(Link)>
          </div>
          <div
            className="column padding-bottom-medium"
          >
            <Connect(Link)
              href="https://www.embl.de/research/tech_transfer/index.html"
            >
              <h5
                className="inline underline"
              >
                Transfer
              </h5>
               
              and development of technology
            </Connect(Link)>
          </div>
          <div
            className="column padding-bottom-medium"
          >
            <h5
              className="inline underline"
            >
              Develop
            </h5>
             
            new instruments and methods
          </div>
          <div
            className="column padding-bottom-medium"
          >
            <h5
              className="inline underline"
            >
              Integrating
            </h5>
             
            life science research in Europe
          </div>
        </div>
        <div
          className="margin-top-xlarge no-underline"
        >
          <h3>
            <Connect(Link)
              className="readmore"
              href="https://embl.org"
            >
              More about EMBL
            </Connect(Link)>
          </h3>
        </div>
      </div>
      <div
        className="columns medium-5"
      >
        <div
          className="large-10 medium-12"
        >
          <h3
            className="inline"
          >
            Six sites
          </h3>
          <p>
            represent EMBL in Europe.
          </p>
        </div>
        <div
          className="row medium-up-2 small-up-2"
        >
          <div
            className="column"
          >
            <h5
              className="inline"
            >
              <Connect(Link)
                href="https://www.embl-barcelona.es/"
              >
                Barcelona
              </Connect(Link)>
            </h5>
            <p>
              Tissue biology and disease modelling
            </p>
          </div>
          <div
            className="column"
          >
            <h5
              className="inline"
            >
              <Connect(Link)
                href="https://www.embl.fr/"
              >
                Grenoble
              </Connect(Link)>
            </h5>
            <p>
              Structural biology
            </p>
          </div>
          <div
            className="column"
          >
            <h5
              className="inline"
            >
              <Connect(Link)
                href="https://www.embl-hamburg.de/"
              >
                Hamburg
              </Connect(Link)>
            </h5>
            <p>
              Structural biology
            </p>
          </div>
          <div
            className="column"
          >
            <h5
              className="inline"
            >
              <Connect(Link)
                href="https://www.embl.de/"
              >
                Heidelberg
              </Connect(Link)>
            </h5>
            <p>
              Main laboratory
            </p>
          </div>
          <div
            className="column"
          >
            <h5
              className="inline"
            >
              <Connect(Link)
                href="https://www.ebi.ac.uk/"
              >
                Hinxton
              </Connect(Link)>
            </h5>
            <p
              className="margin-bottom-none"
            >
              EMBL-EBI: European Bioinformatics Institute
            </p>
          </div>
          <div
            className="column"
          >
            <h5
              className="inline"
            >
              <Connect(Link)
                href="https://www.embl.it/"
              >
                Rome
              </Connect(Link)>
            </h5>
            <p>
              Epigenetics and neurobiology
            </p>
          </div>
        </div>
      </div>
    </div>
  </nav>
</div>
`;

exports[`<EMBLDropdown /> should render as visible 1`] = `
<div
  aria-hidden={false}
  className="masthead-black-bar visible"
>
  <nav
    className="dropdown-pane bottom embl-dropdown"
    id="embl-dropdown"
  >
    <div
      className="row padding-bottom-medium"
    >
      <div
        className="columns padding-top-medium"
      >
        <button
          aria-label="Close alert"
          className="close-button"
          onClick={[Function]}
          type="button"
        >
          <span
            aria-hidden="true"
          >
            ×
          </span>
        </button>
      </div>
      <div
        className="columns medium-7"
      >
        <div
          className="large-10 medium-12"
        >
          <div
            className="margin-bottom-large padding-top-xsmall margin-top-large"
          >
            <h3
              className="no-underline inline"
              style={
                Object {
                  "lineHeight": "1rem",
                }
              }
            >
              <Connect(Link)
                href="https://embl.org"
              >
                EMBL
              </Connect(Link)>
            </h3>
             
            was set up in 1974 as Europe’s flagship laboratory for the life sciences – an intergovernmental organisation with more than 80 independent research groups covering the spectrum of molecular biology:
          </div>
        </div>
        <div
          className="row large-up-3 medium-up-3 small-up-2 no-underline medium-11"
        >
          <div
            className="column padding-bottom-medium"
          >
            <Connect(Link)
              href="https://www.embl.de/research/index.php"
            >
              <h5
                className="inline underline"
              >
                Research:
              </h5>
               
              perform basic research in molecular biology
            </Connect(Link)>
          </div>
          <div
            className="column padding-bottom-medium"
          >
            <Connect(Link)
              href="https://www.embl.de/services/index.html"
            >
              <h5
                className="inline underline"
              >
                Services:
              </h5>
               
              offer vital services to scientists in the member states
            </Connect(Link)>
          </div>
          <div
            className="column padding-bottom-medium"
          >
            <Connect(Link)
              href="https://www.embl.de/training/index.php"
            >
              <h5
                className="inline underline"
              >
                Training
              </h5>
               
              scientists, students and visitors at all levels
            </Connect(Link)>
          </div>
          <div
            className="column padding-bottom-medium"
          >
            <Connect(Link)
              href="https://www.embl.de/research/tech_transfer/index.html"
            >
              <h5
                className="inline underline"
              >
                Transfer
              </h5>
               
              and development of technology
            </Connect(Link)>
          </div>
          <div
            className="column padding-bottom-medium"
          >
            <h5
              className="inline underline"
            >
              Develop
            </h5>
             
            new instruments and methods
          </div>
          <div
            className="column padding-bottom-medium"
          >
            <h5
              className="inline underline"
            >
              Integrating
            </h5>
             
            life science research in Europe
          </div>
        </div>
        <div
          className="margin-top-xlarge no-underline"
        >
          <h3>
            <Connect(Link)
              className="readmore"
              href="https://embl.org"
            >
              More about EMBL
            </Connect(Link)>
          </h3>
        </div>
      </div>
      <div
        className="columns medium-5"
      >
        <div
          className="large-10 medium-12"
        >
          <h3
            className="inline"
          >
            Six sites
          </h3>
          <p>
            represent EMBL in Europe.
          </p>
        </div>
        <div
          className="row medium-up-2 small-up-2"
        >
          <div
            className="column"
          >
            <h5
              className="inline"
            >
              <Connect(Link)
                href="https://www.embl-barcelona.es/"
              >
                Barcelona
              </Connect(Link)>
            </h5>
            <p>
              Tissue biology and disease modelling
            </p>
          </div>
          <div
            className="column"
          >
            <h5
              className="inline"
            >
              <Connect(Link)
                href="https://www.embl.fr/"
              >
                Grenoble
              </Connect(Link)>
            </h5>
            <p>
              Structural biology
            </p>
          </div>
          <div
            className="column"
          >
            <h5
              className="inline"
            >
              <Connect(Link)
                href="https://www.embl-hamburg.de/"
              >
                Hamburg
              </Connect(Link)>
            </h5>
            <p>
              Structural biology
            </p>
          </div>
          <div
            className="column"
          >
            <h5
              className="inline"
            >
              <Connect(Link)
                href="https://www.embl.de/"
              >
                Heidelberg
              </Connect(Link)>
            </h5>
            <p>
              Main laboratory
            </p>
          </div>
          <div
            className="column"
          >
            <h5
              className="inline"
            >
              <Connect(Link)
                href="https://www.ebi.ac.uk/"
              >
                Hinxton
              </Connect(Link)>
            </h5>
            <p
              className="margin-bottom-none"
            >
              EMBL-EBI: European Bioinformatics Institute
            </p>
          </div>
          <div
            className="column"
          >
            <h5
              className="inline"
            >
              <Connect(Link)
                href="https://www.embl.it/"
              >
                Rome
              </Connect(Link)>
            </h5>
            <p>
              Epigenetics and neurobiology
            </p>
          </div>
        </div>
      </div>
    </div>
  </nav>
</div>
`;
